pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1234,6 +1251,49 @@ func (r *ReconcileClusterDeployment) deleteJobOnHashChange(existingJob, generate
 	return newJobNeeded, nil
 }
 
+// cleanupInstallLogPVC will immediately delete the PVC (should it exist) if the cluster was installed successfully, without retries.
+// If there were retries, it will delete the PVC if it has been more than 7 days since the job was completed.
+func (r *ReconcileClusterDeployment) cleanupInstallLogPVC(cd *hivev1.ClusterDeployment, existingJob *batchv1.Job, cdLog log.FieldLogger) error {
```

Comment:
Don't forget to add reconcile tests that verify that the pvc is cleaned up in the scenarios where you expect it to be.